src/oas3/__tests__/bundle.test.ts

```diff
@@ -1197,5 +1203,122 @@ describe('bundleOas3Service', () => {
         $ref: '#/components/headers/SharedHeader1',
       });
     });
+
+    it('operation-level parameter', () => {
+      const res = bundleOas3Service({
+        document: {
+          openapi: '3.0.0',
+          paths: {
+            '/': {
+              get: {
+                operationId: 'the_op_id',
+                parameters: [
+                  {
+                    $ref: '#/components/parameters/Shared1',
+                  },
+                ],
+                responses: { '200': {} },
+              },
+            },
+          },
+          components: {
+            parameters: {
+              Shared1: {
+                $ref: 'target.yaml#/components/parameters/Shared2',
+              },
+            },
+          },
+        },
+      });
+
+      const request = res.operations[0].request;
+      if (!request || isReferenceObject(request)) fail('should be a response');
+      expect(request.unknown).toStrictEqual([
+        {
+          $ref: '#/components/unknownParameters/0',
+        },
+      ]);
+
+      expect(res.components.unknownParameters).toStrictEqual([
+        {
+          $ref: 'target.yaml#/components/parameters/Shared2',
+          key: 'Shared1',
+        },
+      ]);
+    });
+
+    it('path-level parameter', () => {
+      const res = bundleOas3Service({
+        document: {
+          openapi: '3.0.0',
+          paths: {
+            '/': {
+              parameters: [
+                {
+                  $ref: '#/components/parameters/Shared1',
+                },
+              ],
+              get: {
+                operationId: 'the_op_id',
+                responses: { '200': {} },
+              },
+            },
+          },
+          components: {
+            parameters: {
+              Shared1: {
+                $ref: 'target.yaml#/components/parameters/Shared2',
+              },
+            },
+          },
+        },
+      });
+
+      const request = res.operations[0].request;
+      if (!request || isReferenceObject(request)) fail('should be a response');
```

Comment:
FWIW might be good to add https://www.npmjs.com/package/jest-expect-message to avoid that if statement